go/cli/internal/frameworks/adk/python/generator.go

```diff
@@ -0,0 +1,124 @@
+		instruction = string(defaultInstructions)
+	}
+
+	// agent project configuration
+	agentConfig := common.AgentConfig{
+		Name:          agentName,
+		Directory:     projectDir,
+		Framework:     "adk",
+		Language:      "python",
+		Verbose:       verbose,
+		Instruction:   instruction,
+		ModelProvider: modelProvider,
+		ModelName:     modelName,
+	}
+
+	// Use the base generator to create the project
+	if err := g.BaseGenerator.GenerateProject(agentConfig); err != nil {
+		return fmt.Errorf("failed to generate project: %v", err)
+	}
+
+	// Generate project manifest file
+	projectManifest := common.NewProjectManifest(
+		agentConfig.Name,
+		agentConfig.Language,
+		agentConfig.Framework,
+		agentConfig.ModelProvider,
+		agentConfig.ModelName,
+		description,
+	)
+
+	// Save the manifest using the Manager
+	manager := common.NewManifestManager(projectDir)
+	if err := manager.Save(projectManifest); err != nil {
+		return fmt.Errorf("failed to write project manifest: %v", err)
+	}
+
+	// Move agent files from agent/ subdirectory to {agentName} subdirectory
+	agentDir := filepath.Join(projectDir, "agent")
+	if _, err := os.Stat(agentDir); err == nil {
+		// Move all files from agent/ to project subdirectory
+		entries, err := os.ReadDir(agentDir)
+		if err != nil {
+			return fmt.Errorf("failed to read agent directory: %v", err)
+		}
+
+		for _, entry := range entries {
+			if !entry.IsDir() {
+				srcPath := filepath.Join(agentDir, entry.Name())
+				dstPath := filepath.Join(subDir, entry.Name())
+
+				if err := os.Rename(srcPath, dstPath); err != nil {
+					return fmt.Errorf("failed to move %s to %s: %v", srcPath, dstPath, err)
+				}
+			}
+		}
+
+		// Remove the now-empty agent directory
+		if err := os.Remove(agentDir); err != nil {
+			return fmt.Errorf("failed to remove agent directory: %v", err)
+		}
+	}
+
+	fmt.Printf("✅ Successfully created %s project in %s\n", agentConfig.Framework, projectDir)
+	fmt.Printf("📁 Project structure:\n")
+	fmt.Printf("   %s/\n", agentConfig.Name)
+	fmt.Printf("   ├── %s/\n", agentConfig.Name)
+	fmt.Printf("   │   ├── __init__.py\n")
+	fmt.Printf("   │   ├── agent.py\n")
+	fmt.Printf("   │   └── agent-card.json\n")
+	fmt.Printf("   ├── agent.yaml\n")
+	fmt.Printf("   ├── pyproject.toml\n")
+	fmt.Printf("   ├── Dockerfile\n")
+	fmt.Printf("   └── README.md\n")
+	fmt.Printf("\n🚀 Next steps:\n")
+	fmt.Printf("   1. cd %s\n", agentConfig.Name)
+	fmt.Printf("   2. Customize the agent in %s/agent.py\n", agentConfig.Name)
+	fmt.Printf("   3. Build the agent image and push it to the local registry\n")
+	fmt.Printf("      kagent build %s --push\n", agentConfig.Name)
+	fmt.Printf("   4. Deploy the agent to your local cluster\n")
+	fmt.Printf("      kagent deploy %s --api-key-secret <secret-name>\n", agentConfig.Name)
+	fmt.Printf("      Or use --api-key for convenience: kagent deploy %s --api-key <api-key>\n", agentConfig.Name)
+	fmt.Printf("      Support for using a credential file is coming soon\n")
```

Comment:
Currently blocked on procuring the appropriate credentials to test `AnthropicVertexAI` and `VertexAI` which is why I have temporarily removed support for these model providers for now and will add them in a follow up PR.